subcommand/mesh-init/command.go

```diff
@@ -375,6 +378,28 @@ func (c *Command) copyECSBinaryToSharedVolume() error {
 	if err != nil {
 		return err
 	}
+
+	// Dynamic linking (e.g. Cgo) can interfere with reading the executable path
+	// due to the dynamic linker being used to launch the `consul-ecs` binary.
+	// If detected, fall back to searching for the current executable via os.Args and PATH.
+	// c.f. https://go.dev/src/os/executable_path.go
+	if ok, _ := regexp.MatchString("ld-[^.]+\\.so\\.[0-9]+", ex); ok {
+		c.log.Debug("got dynamic linker from os.Executable(), falling back to path search", "found", ex)
+		pathEx, err := exec.LookPath(os.Args[0])
+		if err != nil {
+			return err
+		}
+		pathEx, err = filepath.EvalSymlinks(pathEx)
+		if err != nil {
+			return err
+		}
+		pathEx, err = filepath.Abs(pathEx)
+		if err != nil {
+			return err
+		}
+		ex = pathEx
```

Comment:
Maybe a dumb question, but is there any risk to just doing this in all cases? That way we only have one code path.

Reply:
Not dumb at all! I had the same thought. My main reasons for sticking w/ the fallback were:

- Isolate the fix to what’s broken (FIPS) to avoid regression, esp. since this bug already came from unexpected edge case behavior
- Try to lean on stdlib where we can in hopes of this eventually dying (I’m not certain but I think this could qualify as a de-facto bug in the Linux/procfs implementation of `os.Executable()` that we could report)